.circleci/config.yml

```
@@ -10,88 +10,80 @@ workflows:
       - test-3.7-openmpi
       - test-3.8-mpich
       - test-3.8-openmpi
-      - test-docs
-      - test-linting
 
 default-job: &default-job
-  docker:
-    - image: continuumio/miniconda:latest
   steps:
     - checkout
     - restore_cache:
-        key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}
+        key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}-v2
     - run:
-        name: Install conda environment
-        command: .circleci/install.sh
+        name: Install dependencies in base environment
+        command: conda env update -n base -f ${ENV_FILE}
     - run:
-        name: Running checks
-        command: source activate ${ENV_NAME} && .circleci/${TEST_SCRIPT}
+        name: List packages in the current environment (base)
+        command: conda list
+    - run:
+        name: Running tests
+        command: pytest --junitxml=test-reports/junit.xml --cov=./ --verbose
+    - run:
+        name: Uploading code coverage report
+        command: codecov
+    - store_test_results:
+        path: test-reports
+    - store_artifacts:
+        path: test-reports
     - save_cache:
-        key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}
+        key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}-v2
         paths:
-          - "/opt/conda/envs/${ENV_NAME}/"
           - "/opt/conda/pkgs"
 
 jobs:
 
   test-3.6-mpich:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.6
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.6-mpich.yml"
-      TEST_SCRIPT: "test-unit-coverage.sh"
+      ENV_FILE: ".circleci/env-mpich.yml"
 
   test-3.6-openmpi:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.6
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.6-openmpi.yml"
-      TEST_SCRIPT: "test-unit-only.sh"
+      ENV_FILE: ".circleci/env-openmpi.yml"
       MPIRICAL_MPIRUN: "mpirun --allow-run-as-root"
       OMPI_MCA_rmaps_base_oversubscribe: "1"
 
   test-3.7-mpich:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.7
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.7-mpich.yml"
-      TEST_SCRIPT: "test-unit-only.sh"
+      ENV_FILE: ".circleci/env-mpich.yml"
 
   test-3.7-openmpi:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.7
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.7-openmpi.yml"
-      TEST_SCRIPT: "test-unit-only.sh"
+      ENV_FILE: ".circleci/env-openmpi.yml"
       MPIRICAL_MPIRUN: "mpirun --allow-run-as-root"
       OMPI_MCA_rmaps_base_oversubscribe: "1"
 
   test-3.8-mpich:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.8
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.8-mpich.yml"
-      TEST_SCRIPT: "test-unit-only.sh"
+      ENV_FILE: ".circleci/env-3.8-mpich.yml"
```

@jukent This is the problem!  You need to change this line to read:

```suggestion
      ENV_FILE: ".circleci/env-mpich.yml"
```